### PR TITLE
use rpm's Python bindings to get details of an RPM

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #407

### CODE REVIEW

File: plugins/modules/katello_upload.py [hunk 1]
```diff
@@ -94,6 +121,9 @@ def main():
         ),
     )
 
+    if not HAS_RPM:
+        module.fail_json(msg='The rpm Python module is required', exception=RPM_IMP_ERR)
```
Comment (mdellweg) [inline plugins/modules/katello_upload.py]: As we do this lazy import failing anyway, i think we can be even lazyer and move this in the "if yum" section, or inside the `get_rpm_info`. No need to fail file uploads here.

Comment (evgeni) [reply plugins/modules/katello_upload.py]: `get_rpm_info` has no access to `module`, but I can move that to the `if yum` section, yes.

and need to fix tests. I hate RPM sometimes

**Comment by evgeni** [comment]
@mdellweg mind having a second look at the latest changes? it passes tests now and I need a shower.

**Comment by evgeni** [comment]
Then you didn't look what exactly `rpm-py-installer` does ;-)

Merging before you do ;)

**Comment by mdellweg** [comment]
Ugh! Let's sure hope, that we can still satisfy the dependencies when distributing the collection as a package.

**Comment by evgeni** [comment]
That's why I added it only to the dev requirements. Normal installs shall just use the system python's bindings.

**Pull request merged**